### ISSUE TITLE
Getting results of masks from segmentation and comparing with ground truth 

### ISSUE THREAD

**Issue by swaroop116** [ultralytics/ultralytics#1989]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have done YOLO instance segmentation on NYU indoor dataset. I have been working on my thesis to reduce manual annotation of slam datasets, And I plan on using YOLO and then annotating the images manually to minimize the effort. After YOLO finishes segmentation using the pre-trained model I want to compare the results with the ground truth of the NYU dataset. I have been struck on this for the past few weeks. I went through multiple resources and couldn't find a satisfactory result anywhere.  
Below I am attaching the screenshot of the ground truth of the NYU dataset and also the results I am getting from the YOLO segmentation. 
Here is the ground truth from the NYU dataset
<img width="419" alt="Screenshot 2023-03-30 114825" src="https://user-images.githubusercontent.com/71523052/231569294-020bbb91-d50e-4082-983e-60b9197f03ad.png">
And here is the output I am getting from YOLO
<img width="488" alt="Screenshot 2023-03-30 114942" src="https://user-images.githubusercontent.com/71523052/231569607-5b6e54b3-0c3a-4522-a031-dc997e029fd0.png">
Here is the code I have used for this problem 

<img width="415" alt="Screenshot 2023-03-30 114859" src="https://user-images.githubusercontent.com/71523052/231569761-1dcd8eae-57e5-4858-a239-1aeece4f8e6f.png">

The only unique values are 0 and 1 from the output of the results. 

Please help me in solving this and guide me. You guys are the only ones that can help me with this. Please.

**Comment by Laughing-q** [comment]
@swaroop116 yes the output of yolov8-seg model is a binary mask and the value is either 0 or 1.

**Comment by swaroop116** [comment]
> @swaroop116 yes the output of yolov8-seg model is a binary mask and the value is either 0 or 1.

Is there any way that I can convert the output and compare it with the groundtruth. The groundtruth contains pixelwise annotation where each number denotes the object in that pixel. Is there any way I can convert the output inorder to compare with it.

**Comment by mazatov** [comment]
@swaroop116 , @glenn-jocher  I have the same question. What is the order of the output mask (N, H, W)? How do we know which channel in the output corresponds to which class?
